pkg/ingester/downscale.go

```diff
@@ -0,0 +1,65 @@
+package ingester
+
+import (
+	"net/http"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/services"
+
+	"github.com/grafana/mimir/pkg/util"
+)
+
+// PrepareInstanceRingDownscaleHandler prepares the ingester ring entry for downscaling. It can mark ingester as READONLY or back to ACTIVE.
```

Comment:
This entire comment is referring to `READONLY` ring state, but we moved away from that design, and now only mark ingester as read-only, without changing its state.

Reply:
Updated in https://github.com/grafana/mimir/pull/8956/commits/96afdc1c3d2f62572d1e7f296b0f96552980a9df.